packages/eslint-plugin/src/rules/use-unknown-in-catch-callback-variable.ts

```diff
@@ -0,0 +1,399 @@
+          continue;
+        }
+
+        for (const callSignature of callSignatures) {
+          const firstParam = callSignature.parameters.at(0);
+          if (!firstParam) {
+            // it's not an issue if there's no catch variable at all.
+            continue;
+          }
+
+          let firstParamType = checker.getTypeOfSymbol(firstParam);
+
+          // Deal with a rest arg.
+          const decl = firstParam.valueDeclaration;
+          if (decl != null && ts.isParameter(decl) && decl.dotDotDotToken) {
+            if (checker.isArrayType(firstParamType)) {
+              firstParamType = checker.getTypeArguments(firstParamType)[0];
+            } else if (checker.isTupleType(firstParamType)) {
+              firstParamType = checker.getTypeArguments(firstParamType)[0];
+            } else {
+              // a rest arg that's not an array or tuple should definitely be flagged.
+              return true;
+            }
+          }
+
+          if (!tsUtils.isIntrinsicUnknownType(firstParamType)) {
+            return true;
+          }
+        }
+      }
+
+      return false;
+    }
+
+    /**
+     * If passed an ordinary expression, this will check it as expected.
+     *
+     * If passed a spread element, it treats it as the union of unwrapped array/tuple type.
+     */
+    function checkWhetherShouldFlagArgument(
+      node: TSESTree.Expression | TSESTree.SpreadElement,
+    ): boolean {
+      const argument = services.esTreeNodeToTSNodeMap.get(node);
+      const typeOfArgument = checker.getTypeAtLocation(argument);
+      return isFlaggableHandlerType(typeOfArgument);
+    }
+
+    function checkWhetherShouldFlagMultipleSpreadArgs(
+      argumentsList: TSESTree.CallExpressionArgument[],
+    ): boolean {
+      // One could try to be clever about unpacking fixed length tuples and stuff
+      // like that, but there's no need, since this is all invalid use of `.catch`
+      // anyway at the end of the day. Instead, we'll just check whether any of the
+      // possible args types would violate the rule on its own.
+      return argumentsList.some(argument =>
+        checkWhetherShouldFlagArgument(argument),
+      );
+    }
+
+    function checkWhetherShouldFlagSingleSpreadArg(
+      node: TSESTree.SpreadElement,
+    ): boolean {
+      const spreadArgs = services.esTreeNodeToTSNodeMap.get(node.argument);
+
+      const spreadArgsType = checker.getTypeAtLocation(spreadArgs);
+
+      if (checker.isArrayType(spreadArgsType)) {
+        const arrayType = checker.getTypeArguments(spreadArgsType)[0];
+        return isFlaggableHandlerType(arrayType);
+      }
+
+      if (checker.isTupleType(spreadArgsType)) {
+        const firstType = checker.getTypeArguments(spreadArgsType).at(0);
+        if (!firstType) {
+          // empty spread args. Suspect code, but not a problem for this rule.
+          return false;
+        }
+        return isFlaggableHandlerType(firstType);
+      }
+
+      // otherwise, flag
```

Comment:
[Docs] Nit, I might remove some of these <20 line comments that just restate the code. Up to you.